Sources/Rownd/framework/Automations/AutomationsCoordinator.swift

```diff
@@ -261,27 +250,65 @@ public class AutomationsCoordinator: NSObject, StoreSubscriber {
     }
     
     public func processTrigger(_ automation: RowndAutomation, trigger: RowndAutomationTrigger, state: AutomationStoreState) -> Bool {
-        let lastRunTimestamp = computeLastRunTimestamp(automation: automation, meta: state.user.meta)
+        let lastRunTimestamp = computeLastRunTimestamp(automation: automation, meta: state.user.meta, trigger: trigger)
         switch trigger.type {
         case .time:
-            guard let lastRunTimestamp = lastRunTimestamp else {
-                return true
+            
+            let onceTimeTrigger = automation.triggers.first(where: { $0.type == .timeOnce })
+            let lastRunTimestampOnceTrigger = computeLastRunTimestamp(automation: automation, meta: state.user.meta, trigger: onceTimeTrigger)
+            if let lastRunTimestampOnceTrigger = lastRunTimestampOnceTrigger as? String {
+                // Prevent time trigger if TIME_ONCE hasn't completed yet
+                if (lastRunTimestampOnceTrigger != CompletedAutomationMetaData) {
+                    return false
+                }
             }
-        
+            
             guard let triggerFrequency = stringToSeconds(trigger.value) else {
                 return false
             }
+            
+            guard let lastRunTimestamp = lastRunTimestamp as? Date else {
+                return false
+            }
         
             let dateOfNextPrompt = lastRunTimestamp.addingTimeInterval(Double(triggerFrequency))
-            let currentDate = Clock.now ?? Date()
+            let currentDate = currentDate()
             return currentDate > dateOfNextPrompt
         case .timeOnce:
-            return lastRunTimestamp == nil
+            guard let lastRunTimestamp = lastRunTimestamp else {
+                // Set the intial time if timestamp has not been set
+                setAutomationTime(automation, trigger: trigger)
+                return false
+            }
+            
+            if let lastRunTimestamp = lastRunTimestamp as? String {
+                if (lastRunTimestamp == CompletedAutomationMetaData) {
+                    return false
+                }
+                return false
```

Comment:
This block seems funky with two `return false`'s

Reply:
Yup your right. I removed the 2nd `return false`. It will be caught by the `guard let` check for Date type